tensorflow/lite/micro/examples/hello_world/evaluate.py

```diff
@@ -0,0 +1,142 @@
+# limitations under the License.
+
+import os
+import tensorflow as tf
+from absl import app
+from absl import flags
+import numpy as np
+import matplotlib.pyplot as plt
+from tensorflow.python.platform import resource_loader
+from tflite_micro.tensorflow.lite.micro.python.interpreter.src import \
+  tflm_runtime
+
+_USE_TFLITE_INTERPRETER = flags.DEFINE_bool(
+    'use_tflite', False,
+    'Inference with the TF Lite interpreter instead of the TFLM interpreter')
+
+x_range = 2 * np.pi
+inferences_per_cycle = 1000
+PREFIX_PATH = resource_loader.get_path_to_datafile('')
+
+
+def invoke_tflm_interpreter(input_shape, interpreter, x_value, input_index,
+                            output_index):
+  input_data = np.reshape(x_value, input_shape)
+  interpreter.set_input(input_data, input_index)
+  interpreter.invoke()
+  y_quantized = np.reshape(interpreter.get_output(output_index), -1)[0]
+  return y_quantized
+
+
+def invoke_tflite_interpreter(input_shape, interpreter, x_value, input_index,
+                              output_index):
+  input_data = np.reshape(x_value, input_shape)
+  interpreter.set_tensor(input_index, input_data)
+  interpreter.invoke()
+  tflite_output = interpreter.get_tensor(output_index)
+  y_quantized = np.reshape(tflite_output, -1)[0]
+  return y_quantized
+
+
+# Generate a list of 1000 random floats in the range of 0 to 2*pi.
+def generate_random_input(sample_count=1000):
+  # Generate a uniformly distributed set of random numbers in the range from
+  # 0 to 2π, which covers a complete sine wave oscillation
+  x_values = np.random.uniform(low=0, high=2 * np.pi,
+                               size=sample_count).astype(np.float32)
+  # Shuffle the values to guarantee they're not in order
+  np.random.shuffle(x_values)
+  return x_values
+
+
+# Get the metadata like scales and zero_points from the interpreter input/output
+# details.
+def get_quantization_params(interpreter_io_details):
+  quantize_params = interpreter_io_details.get('quantization_parameters')
+  scale = quantize_params.get('scales')
+  zero_point = quantize_params.get('zero_points')
+  return scale, zero_point
+
+
+# Invoke the tflm interpreter with x_values in the range of [0, 2*PI] and
+# returns the prediction of the interpreter.
+def get_tflm_prediction(hello_world_model_path, x_values):
+  # Create the tflm interpreter
+  tflm_interpreter = tflm_runtime.Interpreter.from_file(hello_world_model_path)
+
+  input_shape = np.array(tflm_interpreter.get_input_details(0).get('shape'))
+
+  y_predictions = np.empty(x_values.size, dtype=np.float32)
+  i = 0
+  for x_value in x_values:
+    y_predictions[i] = invoke_tflm_interpreter(input_shape,
+                                               tflm_interpreter,
+                                               x_value,
+                                               input_index=0,
+                                               output_index=0)
+    i += 1
```

Comment:
Use `enumerate()` here instead. [go/python-tips/023](https://goto.google.com/python-tips/023)

Reply:
Thank you for the suggestion!